gateway/src/apicast/proxy.lua

```diff
@@ -242,7 +242,7 @@ function _M:rewrite(service, context)
     return errors.no_credentials(service)
   end
 
-  local usage, matched_rules = service:get_usage(ngx.req.get_method(), ngx.var.uri)
+  local usage, matched_rules = service:get_usage(ngx.req.get_method(), ngx.var.request_uri)
```

Comment:
AFAIK `$request_uri` includes arguments and `$uri` doesn't.
The [docs](http://nginx.org/en/docs/http/ngx_http_core_module.html) are explicit about `$request_uri`, but not `$uri`.

So, I think the query parameters should be stripped out from `ngx.var.request_uri` first (e.g. `?q=query`).

Reply:
WIP PR, `ngx.var.set_uri()` modified ngx.var.uri and no request.uri, so it's not valid at all ;-)